src/main/java/teammates/ui/controller/Action.java

```diff
@@ -238,10 +238,8 @@ protected AccountAttributes authenticateAndGetNominalUser(UserType loggedInUserT
                 
                 throw new UnauthorizedAccessException("Unregistered user for a page that needs one");
             } else if (isPageNotCourseJoinRelated() && doesRegkeyBelongToUnregisteredStudent() && isUserLoggedIn) {
-                Url redirectUrl = new Url(Const.ViewURIs.LOGOUT)
-                                      .withParam(Const.ParamsNames.NEXT_URL, requestUrl)
-                                      .withUserId(StringHelper.encrypt(account.googleId)); 
-                
+                Url redirectUrl = new Url(student.getRegistrationUrl())
+                                      .withParam(Const.ParamsNames.NEXT_URL, requestUrl);
```

Comment:
hey I think you have to add the UserId parameter so that the admin will be able to masquerade as the student using this redirect